src/main/java/io/omnition/loadgenerator/util/TraceGenerator.java

```diff
@@ -36,36 +38,62 @@ private TraceGenerator(Topology topology) {
 
-    private Span createSpanForServiceRouteCall(ServiceTier serviceTier, String routeName, long startTimeMicros) {
+    private Span createSpanForServiceRouteCall(TagSet parentTagSet, ServiceTier serviceTier, String routeName, long startTimeMicros) {
         String instanceName = serviceTier.instances.get(
                 random.nextInt(serviceTier.instances.size()));
         ServiceRoute route = serviceTier.getRoute(routeName);
-        OperationModel om = serviceTier.getOperationModel(instanceName, routeName);
 
-        // send tags of service and service instance
+        // send tags of serviceTier and serviceTier instance
         Service service = new Service(serviceTier.serviceName, instanceName, new ArrayList<>());
         Span span = new Span();
         span.startTimeMicros = startTimeMicros;
         span.operationName = route.route;
         span.service = service;
-        span.setHttpMethod("GET");
-        span.setHttpUrl("http://" + serviceTier.serviceName + routeName);
-        List<KeyValue> tags = serviceTier.tags.entrySet().stream()
-                .map(t -> KeyValue.ofStringType(t.getKey(), t.getValue())).collect(Collectors.toList());
-        span.tags.addAll(tags);
+
+        // Setup base tags
+        span.setHttpMethodTag("GET");
+        span.setHttpUrlTag("http://" + serviceTier.serviceName + routeName);
+        // Get additional tags for this route, and update with any inherited tags
+        TagSet routeTags = serviceTier.getTagSet(routeName);
+        HashMap<String, Object> tagsToSet = new HashMap<>(routeTags.tags);
+        if (parentTagSet != null && routeTags.inherit != null) {
+            for (String inheritTagKey : routeTags.inherit) {
+                Object value = parentTagSet.tags.get(inheritTagKey);
+                if (value != null) {
+                    tagsToSet.put(inheritTagKey, value);
+                }
+            }
+        }
+
+        // Set the additional tags on the span
+        List<KeyValue> spanTags = tagsToSet.entrySet().stream()
+            .map(t -> {
+                Object val = t.getValue();
+                if (val instanceof String) {
+                    return KeyValue.ofStringType(t.getKey(), (String) val);
+                }
+                if (val instanceof Double) {
+                    return KeyValue.ofLongType(t.getKey(), ((Double) val).longValue());
+                }
+                if (val instanceof Boolean) {
+                    return KeyValue.ofBooleanType(t.getKey(), (Boolean) val);
+                }
+                return null;
+            })
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+        span.tags.addAll(spanTags);
 
         final AtomicLong maxEndTime = new AtomicLong(startTimeMicros);
-        if (random.nextInt(100) < om.errorPercent) {
-            // inject error and terminate trace there
-            span.markError();
+        if (tagsToSet.get(SpanConventions.IS_ERROR_KEY) != null) {
```

Comment:
same as below, perhaps use the isErrorSpan method on Span class as a more generalized implementation.